packages/nextra/src/compile.ts

```diff
@@ -23,15 +23,7 @@ globalThis.__nextra_temp_do_not_use = () => {
   import('./__temp__')
 }
 
-const createCompiler = (mdxOptions: ProcessorOptions): Processor => {
-  const compiler = createProcessor(mdxOptions)
-  compiler.data('headingMeta', {
-    headings: []
-  })
-  return compiler
-}
```

Comment:
since now the compiler is cached it produces hydration error on the client because `headingMeta` are appended on each navigation
![image](https://user-images.githubusercontent.com/7361780/212414590-24635924-76ce-4388-88b1-d3c778e7c435.png)